src/web/app.py

```
@@ -57,153 +59,450 @@
-    - event: citations -> final citations data
-    - event: done -> completion signal
-    - event: error -> error message (followed by done with status: error)
+    # Check length (UUIDs are 36 chars with hyphens)
+    if len(session_id) > 64:
+        logger.warning(f"Session ID too long ({len(session_id)} chars), generating new one")
+        return str(uuid.uuid4())
+
+    # Allow only alphanumeric, hyphens, and underscores
+    # This covers standard UUIDs and common session ID formats
+    if not re.match(r'^[a-zA-Z0-9_-]+$', session_id):
+        logger.warning(f"Session ID contains invalid characters, generating new one")
```

Ruff (0.14.6)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)